### ISSUE TITLE
Could we add an option to model.fit() to keep out N mini-batches per epoch for more robust training?

### ISSUE THREAD

**Issue by bsugerman** [keras-team/keras#10821]
I've been thinking about the extra robustness one gets during training by randomizing what data from the training set goes into each mini-batches, but leaving some percentage of them out during each fit. I believe this may be possible at present by setting `shuffle=True` and manually entering `steps_per_epoch` to be less than the actual number that will be used. 

However, I think this is a useful enough feature that it could deserve its own option. Perhaps we could call it `mutate_percentage` which automatically sets `shuffle=True`, checks to be sure that mini-batches are being used, and either correctly sets `steps_per_epoch`, or ignores thae last `mutate_percentage` of the mini-batches after each reshuffle by adding a couple lines of extra code to the fitting routine.
